src/EditorFeatures/Test2/InlineParameterNameHints/AbstractInlineParamNameHintsTests.vb

```diff
@@ -0,0 +1,62 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
+Imports Microsoft.CodeAnalysis.Editor.Shared.Extensions
+Imports Microsoft.CodeAnalysis.Editor.Shared.Utilities
+Imports Microsoft.CodeAnalysis.Editor.Tagging
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces
+Imports Microsoft.CodeAnalysis.Shared.TestHooks
+Imports Microsoft.VisualStudio.Text
+
+Namespace Microsoft.CodeAnalysis.Editor.UnitTests.InlineParameterNameHints
+
+    <[UseExportProvider]>
+    Public MustInherit Class AbstractInlineParamNameHintsTests
+
+        Protected Async Function VerifyParamHints(test As XElement, Optional optionIsEnabled As Boolean = True) As Tasks.Task
+            Await VerifyParamHints(test, optionIsEnabled, outOfProcess:=False)
+            Await VerifyParamHints(test, optionIsEnabled, outOfProcess:=True)
+        End Function
+
+        Private Async Function VerifyParamHints(test As XElement, optionIsEnabled As Boolean, outOfProcess As Boolean) As Tasks.Task
+            Using workspace = TestWorkspace.Create(test)
+                WpfTestRunner.RequireWpfFact($"{NameOf(AbstractInlineParamNameHintsTests)}.{NameOf(Me.VerifyParamHints)} creates asynchronous taggers")
+
+                Dim tagProducer = New InlineParamNameHintsDataTaggerProvider(
+                    workspace.ExportProvider.GetExportedValue(Of IThreadingContext),
+                    AsynchronousOperationListenerProvider.NullProvider,
+                    workspace.GetService(Of IForegroundNotificationService))
+
+                Dim hostDocument = workspace.Documents.Single(Function(d) d.CursorPosition.HasValue)
+                Dim caretPosition = hostDocument.CursorPosition.Value
```

Comment:
Why do we need a caret position for these tests? The tagger isn't using a position, right?